examples/demo/app/src/references/mod.rs

```diff
@@ -24,12 +24,27 @@ impl ReferenceService {
         }
     }
 
+    pub fn add<'a>(&mut self, v: u32) -> &'a u32 {
```

Comment:
Would it be more illustrative if:
1. We make the `ReferenceCount` implementing the `Default` trait
2. Pass it from the outside as `&'a mut ReferenceCount`
3. Hold it as a static in program itself